crates/node/src/shell/finalize_block.rs

```diff
@@ -1710,6 +1711,27 @@ mod test_finalize_block {
         });
     }
 
+    /// Test the correct transition to a new masp epoch
+    #[test]
+    fn test_masp_epoch_progression() {
+        let (mut shell, _broadcaster, _, _eth_control) = setup();
+
+        let masp_epoch_multiplier =
+            namada::ledger::parameters::read_masp_epoch_multiplier_parameter(
+                &shell.state,
+            )
+            .unwrap();
+
+        assert_eq!(shell.state.get_block_epoch().unwrap(), Epoch::default());
+
+        for _ in 1..masp_epoch_multiplier {
+            shell.start_new_epoch(None);
+            assert!(!shell.state.is_masp_new_epoch(true).unwrap());
+        }
+        shell.start_new_epoch(None);
```

Comment:
the assertion inside the loop won't be performed when `masp_epoch_multiplier` has a value of 1, because of the open range (`1..masp_epoch_multiplier`). maybe add the same assertion just before the loop starts
```rs
assert!(!shell.state.is_masp_new_epoch(true).unwrap());
```

Reply:
Actually if the multiplier is indeed one the assertion will fail because in that case we would indeed start a new masp epoch too